mlflow/models/evaluation/base.py

```diff
@@ -906,9 +1135,25 @@ def scatter_plot(eval_df, builtin_metrics, artifacts_dir):
     elif isinstance(model, PyFuncModel):
         pass
     else:
-        raise ValueError(
-            "The model argument must be a string URI referring to an MLflow model or "
-            "an instance of `mlflow.pyfunc.PyFuncModel`."
+        raise MlflowException(
+            message="The model argument must be a string URI referring to an MLflow model or "
+            "an instance of `mlflow.pyfunc.PyFuncModel`.",
+            erorr_code=INVALID_PARAMETER_VALUE,
+        )
+
+    if isinstance(baseline_model, str):
+        baseline_model = mlflow.pyfunc.load_model(baseline_model)
+    elif baseline_model is not None:
+        raise MlflowException(
+            message="The baseline model argument must be a string URI referring to an MLflow model",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+    elif _model_validation_contains_model_comparison(validation_thresholds):
+        raise MlflowException(
+            message="The baseline model argument must be a string URI referring to "
+            "an MLflow model when model comparison thresholds  "
+            "(min_absolute_change, min_relative_change) are specified.",
+            error_code=INVALID_PARAMETER_VALUE,
```

Comment:
When I ran `mlflow.evaluate()` and specified validation thresholds without specifying a baseline model, the error message was:

```
mlflow.exceptions.MlflowException: The baseline model argument must be a string URI referring to an MLflow model when model comparison thresholds  (min_absolute_change, min_relative_change) are specified
```

This is kind of confusing, as it makes it seem like the issue is a type error when, in fact, the issue is that the baseline was never specified. Can we distinguish between these cases (wrong type vs absent) and raise different errors?